src/main/java/com/thealgorithms/dynamicprogramming/LongestArithmeticSubsequence.java

```diff
@@ -0,0 +1,45 @@
+package com.thealgorithms.dynamicprogramming;
+
+import java.util.HashMap;
+
+final class LongestArithmeticSubsequence {
+
+    private LongestArithmeticSubsequence() {
+    }
+
+    /**
+     * Returns the length of the longest arithmetic subsequence in the given array.
+     *
+     * A sequence seq is arithmetic if seq[i + 1] - seq[i] are all the same value
+     * (for 0 <= i < seq.length - 1).
+     *
+     * @param nums the input array of integers
+     * @return the length of the longest arithmetic subsequence
+     */
+    public static int getLongestArithmeticSubsequenceLength(int[] nums) {
+        // If the array is empty or has only one element, return its length.
+        if (nums == null || nums.length == 0) {
```

Comment:
What do you think about throwing an exception if `nums` is `null`?

Please add a test, with the input being `null`.

Reply:
agreed, throwing an exception for the null case will be better to handle the invalid input explicitly